src/dfx/src/lib/models/canister_id_store.rs

```diff
@@ -93,31 +102,21 @@ impl CanisterIdStore {
             .map(|(canister_name, _)| canister_name)
     }
 
-    #[context("Failed to load ids from storage at {}.", path.to_string_lossy())]
-    pub fn load_ids(path: &Path) -> DfxResult<CanisterIds> {
-        let content = std::fs::read_to_string(path)
-            .with_context(|| format!("Cannot read from file at '{}'.", path.display()))?;
-        serde_json::from_str(&content)
-            .with_context(|| format!("Cannot decode contents of file at '{}'.", path.display()))
-    }
-
-    pub fn save_ids(&self) -> DfxResult {
+    pub fn save_ids(&self) -> Result<(), CanisterStoreIdError> {
         let path = self
             .path
             .as_ref()
             .unwrap_or_else(|| {
                 // the only callers of this method have already called Environment::get_config_or_anyhow
                 unreachable!("Must be in a project (call Environment::get_config_or_anyhow()) to save canister ids")
             });
-        let content =
-            serde_json::to_string_pretty(&self.ids).context("Failed to serialize ids.")?;
-        let parent = path.parent().unwrap();
+        let content = dfx_core::json::pretty_print(&self.ids)?;
+        let parent = dfx_core::fs::parent(path)?;
```

Comment:
save_json_file pretty-prints

```suggestion
        dfx_core::fs::composite::ensure_parent_dir_exists(path)?;
        dfx_core::json::save_json_file(path, content)?;
```